tests/test_subparser.py

```diff
@@ -251,12 +252,13 @@ def test_flag(self, capsys):
         assert "unrecognized arguments: --flag" in captured.err
 
     def test_command_args(self):
-        config = parse(self.Config, ["Command1", "--cmd-a", "12"])
+        config = parse(self.Config, ["--a", "12", "Command1", "--a", "12"])
+        assert config.a == 12.0
```

Comment:
```suggestion
        config = parse(self.Config, ["--a", "11", "Command1", "--a", "12"])
        assert config.a == 11.0
```

Just to make it clear it actually assigns the value from the first encounter of --a :+1: